### PR TITLE
use a single SQL query for any number of json-api query pairs

### PR DESCRIPTION
**Pull request by S11001001**
Since #8226 we have been doing multiple SQL queries and deduplicating them on the Scala side if there were multiple queries for a given template ID. Instead, we can rerun the predicate in the SELECT clause and append to a list of query numbers instead, making it so that we don't have to do any deduplication in Scala, as briefly imagined in #9278.

This is better for complex multiqueries, of course, and also makes it possible to consider streaming database results in _all_ cases in the future.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Nice, looks sensible albeit a bit intricate but hopefully tests catch the subtleties.

Thanks!

File: ledger-service/db-backend/src/test/scala/com/digitalasset/http/dbbackend/QueriesSpec.scala [hunk 1]
```diff
@@ -3,33 +3,75 @@
...
+            sql"CASE WHEN ( tpid = ${66L}) THEN ${1}||''  WHEN ( tpid = ${55L}) THEN ${0}||''  ELSE NULL END ",
+        ),
+      ),
+      (
+        Seq(
+          (SurrogateTpId(55L), fr"foo") -> 0,
+          (SurrogateTpId(55L), fr"bar") -> 1,
+          (SurrogateTpId(66L), fr"baz") -> 2,
+        ),
+        Reordered(
```
Comment (cocreature) [inline ledger-service/db-backend/src/test/scala/com/digitalasset/http/dbbackend/QueriesSpec.scala:44]: Should we just use an ordered map instead? This really shouldn’t be our performance bottleneck and that seems like a much nicer solution than hardcoding the hash order.

Comment (S11001001) [reply ledger-service/db-backend/src/test/scala/com/digitalasset/http/dbbackend/QueriesSpec.scala:44]: Yeah, we can, I'm just being stubborn about my instances.

Comment (S11001001) [reply ledger-service/db-backend/src/test/scala/com/digitalasset/http/dbbackend/QueriesSpec.scala]: fc7db44d36b

Comment (cocreature) [reply ledger-service/db-backend/src/test/scala/com/digitalasset/http/dbbackend/QueriesSpec.scala]: nice, thank you

**Pull request merged**